src/core-sdk-tasks/DownloadFile.cs

```diff
@@ -14,12 +16,25 @@ public class DownloadFile : Task
         [Required]
         public string Uri { get; set; }
 
+        /// <summary>
+        /// If this field is set and the task fail to download the file from `Uri` it will try
```

Comment:
Description is a little wrong, it only falls back to private URI if it hits a 404, not simply on failure.